src/core/qgsapplication.cpp

```diff
@@ -289,6 +292,9 @@ void QgsApplication::init( QString profileFolder )
 
 QgsApplication::~QgsApplication()
 {
+  if ( mApplicationMembers->mActiveProject && mApplicationMembers->mActiveProject->parent() == this )
+    delete mApplicationMembers->mActiveProject;
```

Comment:
Just asking: is there any chance to have automatic memory management such as std::unique_ptr for `mActiveProject` ?

Reply:
No, mixing QObject parents and unique_ptrs is a bad idea